projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -399,6 +401,29 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
     }
   }
 
+  private getMaskedValue(): string {
+    let mask = this.emptyMask;
+    for (const part of this._inputDateParts) {
+      if (part.type === DatePart.Literal) { continue; }
+      const partLength = part.format.length;
+      let targetValue = this.getPartValue(part.type, partLength);
+      if (part.type === DatePart.Month) {
```

Comment:
here we ensure month part has leading 0 in it is less than 10
what about day, minute and second parts - should we do the same for them?